### PR TITLE
v0.14.1

### PR DESCRIPTION
**Pull request by CamDavidsonPilon**
#### 0.14.1
 - fixed bug with using weights and strata in `CoxPHFitter`
 - fixed bug in using non-integer weights in `KaplanMeierFitter`
 - Performance optimizations in `CoxPHFitter` for up to 40% faster completion of `fit`.
    - even smarter `step_size` calculations for iterative optimizations. 
    - simple code optimizations & cleanup in specific hot spots.
 - Performance optimizations in `AalenAdditiveFitter` for up to 50% faster completion of `fit` for large dataframes, and up to 10% faster for small dataframes.

### CODE REVIEW

File: lifelines/fitters/aalen_additive_fitter.py [hunk 1]
```diff
@@ -218,8 +217,9 @@ def _fit_static(self, dataframe, duration_col, event_col=None,
         self.durations = T
         self.event_observed = C
         self._compute_confidence_intervals()
+
         self.score_ = concordance_index(self.durations,
-                                        self.predict_median(dataframe).values.ravel(),
+                                        self.predict_cumulative_hazard(dataframe).values[-1,:].ravel(),
```
Comment (CamDavidsonPilon) [inline lifelines/fitters/aalen_additive_fitter.py]: not sure I want to keep this

File: lifelines/fitters/coxph_fitter.py [hunk 2]
```diff
@@ -84,6 +88,8 @@ def fit(self, df, duration_col, event_col=None,
             self, with additional properties: hazards_
 
         """
+        #import pdb
+        #pdb.set_trace()
```
Comment (CamDavidsonPilon) [inline lifelines/fitters/coxph_fitter.py]: rm

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
 ### Changelogs
 
+#### 0.15.0
```
Comment (CamDavidsonPilon) [inline CHANGELOG.md]: I think I'll make this 0.14.1 and remove any API changes.

File: lifelines/fitters/aalen_additive_fitter.py [hunk 4]
```diff
@@ -141,14 +141,16 @@ def _fit_static(self, dataframe, duration_col, event_col=None,
...
-        pass_for_numeric_dtypes_or_raise(df)
+        self._check_values(df, T, C)
 
         ix = T.argsort()
         T, C = T.iloc[ix], C.iloc[ix]
 
         del df[duration_col]
         n, d = df.shape
         columns = df.columns
+        df = df.astype(float)
```
Comment (springcoil) [inline lifelines/fitters/aalen_additive_fitter.py]: I'm not sure I follow exactly what has caused the performance improvements. Is it the float? Or is it something else?

Comment (CamDavidsonPilon) [reply lifelines/fitters/aalen_additive_fitter.py]: It was the changes here: https://github.com/CamDavidsonPilon/lifelines/pull/433/files#diff-81aff105858474f333c6fc40c9edae5eR650

File: lifelines/fitters/aalen_additive_fitter.py (hunk 1 above)
Comment (CamDavidsonPilon) [reply lifelines/fitters/aalen_additive_fitter.py]: this changes the API, hence a version upgrade.

Comment (springcoil) [reply lifelines/fitters/aalen_additive_fitter.py]: Hmm I think it's useful but changing the version is a big change.

File: lifelines/fitters/aalen_additive_fitter.py (hunk 4 above)
Comment (springcoil) [reply lifelines/fitters/aalen_additive_fitter.py]: I just found it. So the matrix is always positive definite, that makes sense.

**Pull request closed without merging**